lib/DxcSupport/dxcapi.extval.cpp

```diff
@@ -38,21 +37,20 @@ HRESULT DxcDllExtValidationSupport::InitializeInternal(LPCSTR dllName,
     return S_OK;
   }
 
-  std::string DllPathStr(EnvVarVal);
-  DxilDllPath = DllPathStr;
-  std::filesystem::path DllPath(DllPathStr);
+  DxilDllPath = std::string(EnvVarVal);
+  std::filesystem::path DllPath(DxilDllPath);
 
   // Check if path is absolute and exists
   if (!DllPath.is_absolute() || !std::filesystem::exists(DllPath)) {
-    return S_OK;
+    // ERROR_DLL_INIT_FAILED
+    return (HRESULT)1114L;
```

Comment:
This isn't an HRESULT, it's an error code (how can I tell? HRESULTs are have the high-bit set if they're an actual error condition and so would always show up as negative numbers when you write them like this).  See https://godbolt.org/z/rxKjr9jTT

Is there actually a test that's testing that the call to Initialize fails?

You could consider using `HRESULT_FROM_WIN32(1114L)`, but you're still using a pretty obscure error code, and one that actually has quite a different meaning - it suggests something failed in the OS's DLL loader rather than something failing in normal usermode logic.

I think `E_INVALIDARG` actually conveys the right sort of thing here - plus if we ever see it in this code path, we'll know what to look for.

Reply:
And yes, initialization is tested on line 4261.